ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Worker.java

```diff
@@ -390,7 +390,7 @@ protected Boolean findNextCompactionAndExecute(boolean computeStats) {
         }
       }
       ci = CompactionInfo.optionalCompactionInfoStructToInfo(msc.findNextCompact(workerName, runtimeVersion));
-      if ((runtimeVersion != null || ci.initiatorVersion != null) && !runtimeVersion.equals(ci.initiatorVersion)) {
+      if (ci != null && (runtimeVersion != null || ci.initiatorVersion != null) && !runtimeVersion.equals(ci.initiatorVersion)) {
```

Comment:
Hi Anti, I placed by mistake version check above the ci null check part, could you please move it below the null check:
if (ci == null) {
        return false;
}

Reply:
Sure, done.